### DISCUSSION TITLE
Network communication auto encryption, or upgrade golang to 1.20, or compile wireguard by golang 1.18

### DISCUSSION THREAD

**Discussion by AlexNebula** [occlum/occlum#1435] [Ideas]
It seems that Occlum does not provide network communication auto encryption feature, I have one based on wireguard, but wireguard uses new features of golang 1.20. So, either Occlum provides network communication auto encryption feature (just like its file I/O auto encryption feature), or Occlum supports golang 1.20 so that I can integrate my network communication auto encryption module into Occlum, or there is a solution to compile wireguard by Occlum golang 1.18

**Comment by volcano0dr** [comment]
Recently Occlum supported golang 1.20.
You can clone the [ago1.20.12_for_occlum](https://github.com/occlum/go/tree/ago1.20.12_for_occlum) repo into your own docker development container and then compile and install it. You can refer to the steps in [build.sh](https://github.com/occlum/occlum/blob/master/tools/toolchains/golang/build.sh).
